tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -774,114 +774,53 @@ void test_lte_lc_psm_get_null_fail(void)
-		"%XMONITOR: 1,\"Operator\",\"OP\",\"20065\",\"002F\",7,20,\"0012BEEF\","
-		"334,6200,66,44";
-	__cmock_nrf_modem_at_cmd_ExpectAndReturn(NULL, 0, "AT%%XMONITOR", 0);
-	__cmock_nrf_modem_at_cmd_IgnoreArg_buf();
-	__cmock_nrf_modem_at_cmd_IgnoreArg_len();
-	__cmock_nrf_modem_at_cmd_ReturnArrayThruPtr_buf(
-		(char *)xmonitor_resp, sizeof(xmonitor_resp));
-
-	ret = lte_lc_psm_get(&tau, &active_time);
-	TEST_ASSERT_EQUAL(-EBADMSG, ret);
-}
-
-void test_lte_lc_psm_get_badmsg_no_active_time_fail(void)
-{
-	int ret;
-	int tau;
-	int active_time;
-
-	static const char xmonitor_resp[] =
-		"%XMONITOR: 1,\"Operator\",\"OP\",\"20065\",\"002F\",7,20,\"0012BEEF\","
-		"334,6200,66,44,\"\"";
-	__cmock_nrf_modem_at_cmd_ExpectAndReturn(NULL, 0, "AT%%XMONITOR", 0);
-	__cmock_nrf_modem_at_cmd_IgnoreArg_buf();
-	__cmock_nrf_modem_at_cmd_IgnoreArg_len();
-	__cmock_nrf_modem_at_cmd_ReturnArrayThruPtr_buf(
-		(char *)xmonitor_resp, sizeof(xmonitor_resp));
-
-	ret = lte_lc_psm_get(&tau, &active_time);
-	TEST_ASSERT_EQUAL(-EBADMSG, ret);
-}
-
 void test_lte_lc_psm_get_badmsg_no_tau_ext_fail(void)
 {
 	int ret;
 	int tau;
 	int active_time;
 
-	static const char xmonitor_resp[] =
-		"%XMONITOR: 1,\"Operator\",\"OP\",\"20065\",\"002F\",7,20,\"0012BEEF\","
-		"334,6200,66,44,\"\",\"00000101\"";
-	__cmock_nrf_modem_at_cmd_ExpectAndReturn(NULL, 0, "AT%%XMONITOR", 0);
-	__cmock_nrf_modem_at_cmd_IgnoreArg_buf();
-	__cmock_nrf_modem_at_cmd_IgnoreArg_len();
-	__cmock_nrf_modem_at_cmd_ReturnArrayThruPtr_buf(
-		(char *)xmonitor_resp, sizeof(xmonitor_resp));
+	__mock_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT%XMONITOR",
+		"%%XMONITOR: %*u,%*[^,],%*[^,],%*[^,],%*[^,],%*d,%*u,%*[^,],%*u,%*u,%*d,%*d,%*[^,],\"%8[^\"]\",\"%8[^\"]\",\"%8[^\"]\"",
+		1);
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("11100000");
 
 	ret = lte_lc_psm_get(&tau, &active_time);
 	TEST_ASSERT_EQUAL(-EBADMSG, ret);
 }
 
-void test_lte_lc_psm_get_badmsg_no_legacy_tau_fail(void)
+void test_lte_lc_psm_get_inval_legacy_tau_len_not_8_fail(void)
 {
 	int ret;
 	int tau;
 	int active_time;
 
-	static const char xmonitor_resp[] =
-		"%XMONITOR: 1,\"Operator\",\"OP\",\"20065\",\"002F\",7,20,\"0012BEEF\","
-		"334,6200,66,44,\"\",\"00000101\",\"00010011\"";
-	__cmock_nrf_modem_at_cmd_ExpectAndReturn(NULL, 0, "AT%%XMONITOR", 0);
-	__cmock_nrf_modem_at_cmd_IgnoreArg_buf();
-	__cmock_nrf_modem_at_cmd_IgnoreArg_len();
-	__cmock_nrf_modem_at_cmd_ReturnArrayThruPtr_buf(
-		(char *)xmonitor_resp, sizeof(xmonitor_resp));
+	__mock_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT%XMONITOR",
+		"%%XMONITOR: %*u,%*[^,],%*[^,],%*[^,],%*[^,],%*d,%*u,%*[^,],%*u,%*u,%*d,%*d,%*[^,],\"%8[^\"]\",\"%8[^\"]\",\"%8[^\"]\"",
+		3);
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("11100000");
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("00010011");
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("001001");
 
 	ret = lte_lc_psm_get(&tau, &active_time);
-	TEST_ASSERT_EQUAL(-EBADMSG, ret);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
```

Comment:
While at it, you could modify the implementation to return `-EBADMSG` when parsing fails. I think the only case when `-EINVAL` should be returned is when one of the function parameters is NULL:

` * @retval -EINVAL if input argument was invalid.`

Reply:
This is something I thought at some point, i.e., what to do with a return value that is passed from another function (`parse_psm()` in this case) but let it be as it's been in many places.
I'll change `lte_lc_psm_get()` return value to `-EBADMSG` when `parse_psm()` fails as its only possible return value is `-EINVAL`.